src/ca/sigmsg.rs

```diff
@@ -402,20 +417,24 @@ impl SignedMessageCrl {
         when: Time
     ) -> Result<(), ValidationError> {
         if self.tbs.signature != *self.signed_data.signature().algorithm() {
-            return Err(ValidationError)
+            return Err(VerificationError::new(
+                "CRL signature algorithm mismatch"
+            ).into())
         }
         self.signed_data.verify_signature(
             issuer_key
-        )?;
+        ).map_err(VerificationError::from)?;
         self.tbs.validate(issuer_key, when)
     }
 
     /// Returns whether the given serial number is on this revocation list.
-    fn validate_not_revoked(
+    fn verify_not_revoked(
         &self, id_cert: &IdCert
-    ) -> Result<(), ValidationError> {
+    ) -> Result<(), VerificationError> {
         if self.tbs.revoked_certs.contains(id_cert.serial_number()) {
-            Err(ValidationError)
+            Err(VerificationError::new(
+                "revoked certificate"
```

Comment:
Should this text mention CRL? The ones above mention for example CRL or signed object.

Reply:
pushed proposal